src/devices/bus/msx_slot/slot.h

```diff
@@ -25,22 +25,32 @@ class msx_internal_slot_interface
 	// configuration helpers
 	template <typename T> void set_memory_space(T &&tag, int spacenum) { m_mem_space.set_tag(std::forward<T>(tag), spacenum); }
 	template <typename T> void set_io_space(T &&tag, int spacenum) { m_io_space.set_tag(std::forward<T>(tag), spacenum); }
-	void set_start_address(uint32_t start_address) { m_start_address = start_address; m_end_address = m_start_address + m_size; }
-	void set_size(uint32_t size) { m_size = size; m_end_address = m_start_address + m_size; }
+	template <typename T> void set_maincpu(T &&tag) { m_maincpu.set_tag(std::forward<T>(tag)); }
+	void set_start_address(u32 start_address) { m_start_address = start_address; m_end_address = m_start_address + m_size; }
+	void set_size(u32 size) { m_size = size; m_end_address = m_start_address + m_size; }
 
-	virtual uint8_t read(offs_t offset) { return 0xFF; }
-	virtual void write(offs_t offset, uint8_t data) { }
+	void install(memory_view::memory_view_entry *page0, memory_view::memory_view_entry *page1, memory_view::memory_view_entry *page2, memory_view::memory_view_entry *page3);
 
 	address_space &memory_space() const { return *m_mem_space; }
 	address_space &io_space() const { return *m_io_space; }
+	cpu_device &maincpu() const { return *m_maincpu; }
+	bool page_configured(int i) { return bool(m_page[i]); }
+	memory_view::memory_view_entry *page(int i) {
```

Comment:
We generally put the opening brace on the beginning of the line following the function signature, even in K&R files.

Reply:
Thanks for the comments so far